controllers/argocd_controller.go

```diff
@@ -119,26 +132,34 @@ func (r *ReconcileArgoCDRoute) Reconcile(ctx context.Context, request reconcile.
 		}
 		return reconcile.Result{}, err
 	}
+
 	reqLogger.Info("Route found for argocd-server", "Route.Host", argoCDRoute.Spec.Host)
 
-	argocCDRouteURL := fmt.Sprintf("https://%s", argoCDRoute.Spec.Host)
+	argoCDRouteURL := fmt.Sprintf("https://%s", argoCDRoute.Spec.Host)
 
-	consoleLink := newConsoleLink(argocCDRouteURL, "Cluster Argo CD")
+	consoleLink := newConsoleLink(argoCDRouteURL, "Cluster Argo CD")
 
 	found := &console.ConsoleLink{}
 	err = r.Client.Get(ctx, types.NamespacedName{Name: consoleLink.Name}, found)
+
 	if err != nil {
 		if errors.IsNotFound(err) {
-			reqLogger.Info("Creating a new ConsoleLink", "ConsoleLink.Name", consoleLink.Name)
-			return reconcile.Result{}, r.Client.Create(ctx, consoleLink)
+			if !isConsoleLinkDisabled() {
+				reqLogger.Info("Creating a new ConsoleLink", "ConsoleLink.Name", consoleLink.Name)
+				return reconcile.Result{}, r.Client.Create(ctx, consoleLink)
+			}
 		}
 		reqLogger.Error(err, "Failed to create ConsoleLink", "ConsoleLink.Name", consoleLink.Name)
 		return reconcile.Result{}, err
 	}
+	if isConsoleLinkDisabled() {
+		reqLogger.Info("consoleLink deleted", "Route.Host", argoCDRoute.Spec.Host)
+		return reconcile.Result{}, r.deleteConsoleLinkIfPresent(ctx, reqLogger)
+	}
 
-	if found.Spec.Href != argocCDRouteURL {
+	if found.Spec.Href != argoCDRouteURL {
```

Comment:
this (if found) block should be enclosed in the "else" clause of the if block above. You don't update the the link if you just removed it.